public/app/core/components/AppChrome/MegaMenu/MegaMenu.tsx

```diff
@@ -39,6 +39,28 @@ export const MegaMenu = memo(
       .filter((item) => item.id !== 'profile' && item.id !== 'help')
       .map((item) => enrichWithInteractionTracking(item, state.megaMenuDocked));
 
+    if (config.featureToggles.pinNavItems) {
+      const bookmarksItem = findByUrl(navItems, '/bookmarks');
+      if (bookmarksItem) {
+        // Add children to the bookmarks section
+        bookmarksItem.children = pinnedItems.reduce((acc: NavModelItem[], url) => {
+          const item = findByUrl(navItems, url);
```

Comment:
It looks like some nav items have the same URLs. E.g., if I bookmark `Reporting`, `Reports` also appears as bookmarked, altho it's not shown in the `Bookmarks` section. Not sure if it's a big deal or if there are many other cases like this tho. 

![Screenshot 2024-08-15 at 13 38 37](https://github.com/user-attachments/assets/b7ea7efc-73ee-4f31-92ab-f85c371c1061)

Reply:
Ah yes I noticed this! This happens because we have changed from storing `id` to storing `url` in the bookmarks. IMO this isn't a big deal... Not sure how common it is and it seems more like a problem with how that navigation section is setup, there's already a problem even without bookmarks that if you click "Reporting" it highlights "Reports" as the active item exactly because the url is the same, so don't think its worth us trying to fix 🤷